lib/mapObjects/CGMarket.cpp

```diff
@@ -284,13 +285,13 @@ std::vector<int> CGBlackMarket::availableItemsIds(EMarketMode::EMarketMode mode)
 
 void CGBlackMarket::newTurn(CRandomGenerator & rand) const
 {
-	if(cb->getDate(Date::DAY_OF_MONTH) != 1) //new month
-		return;
-
-	SetAvailableArtifacts saa;
-	saa.id = id.getNum();
-	cb->pickAllowedArtsSet(saa.arts, rand);
-	cb->sendAndApply(&saa);
+	if(VLC->modh->settings.BLACK_MARKET_MONTHLY_ARTIFACTS_CHANGE && cb->getDate(Date::DAY_OF_MONTH) == 1) //new month and feature enabled
```

Comment:
I would suggest to use dedicated if instead:
```
if(!VLC->modh->settings.BLACK_MARKET_MONTHLY_ARTIFACTS_CHANGE)
	return;
```
So it's easier to understand is some very special hardcoded option and not just one of parameters.